### ISSUE TITLE
Failed to write to workspace

### ISSUE THREAD

**Issue by JackWesleyNelson** [slynch8/10x#782]
![gYmNS3cY4f](https://user-images.githubusercontent.com/8739573/155542001-a867f6b0-4cee-4a38-8e35-5539ef54a19b.png)
When editing the build settings for a workspace at  C:Programming/MyWinProgramming/Programs/ExampleProgram/ExampleProgram.10x I get a failed to write workspace for the workspace lower down in the tree at C:Programming/MyWinProgramming/MyWinProgramming.10x saying access is denied.

10x version: 0.31.9

**Comment by slynch8** [comment]
Are you using perforce? Current workspaces are not checked out automatically

**Comment by JackWesleyNelson** [comment]
No, I'm not using any form of source/version control on any of the files in the MyWinProgramming directory (or subdirectories).

**Comment by slynch8** [comment]
is the file read-only?

**Comment by JackWesleyNelson** [comment]
No, neither of the workspace files are set as read only, they both have full permissions aside from "Special Permission", as both the current user and as administrator, which 10x was currently running as (disabling run as administrator results in the same behavior).

**Comment by JackWesleyNelson** [comment]
Looks Like the folder itself is resetting to read-only, which I didn't realize, presumably something to do with it being directly in the C: directory, and not something to do with 10x itself. My mistake, if the behavior occurs elsewhere where the folder doesn't reset to read only I'll update the issue.

**Comment by JackWesleyNelson** [comment]
I made the mistake of thinking that the checkbox being filled in meant it was read-only, rather than explicitly checked. So, my initial assumption that none of the file or folder attributes were read-only was actually correct (As far as I can tell from inspecting the attributes and security permissions at least). Moving the folder to my documents folder and confirming that none of the files are read-only resulted in the same behavior. I don't believe I ran into the issue in version 0.31.3, and only really noticed it sometime after updating to at least 0.31.4. If there is a way for me to download a specific version I can check if that allows me to save the workspace properly or not for the current project in question.

**Comment by slynch8** [comment]
all the old versions are here
https://10xeditor.com/prev_versions.htm
You'll need to uninstall the current version before installing an older one. But I don't think anything has changed recently that would cause this.
Does it still happen if you create a new workspace in the root of C or another drive?

**Comment by JackWesleyNelson** [comment]
So some variants of creating new workspaces that are nested seemed to work while some didn't, but I believe I manage to replicate the steps that created the original problem.
-Create RootFolder folder
-Create RootWorkspace workspace inside RootFolder
-Create Programs folder in RootFolder folder
-Create ExampleProgram folder in Programs folder
-Exclude ExampleProgram folder in RootWorkspace workspace
-Right click Programs folder and click New Workspace
-Create ExampleProgramWorkspace inside ExampleProgram folder
-Change build command of ExampleProgramWorkspace
-Save Workspace
-Close 10x
-Open 10x
-Build command of ExampleProgramWorkspace should be blank
-Change build command of ExampleProgramWorkspace again
-Save Workspace again
-Failed to Write, Access is denied dialog should appear

**Comment by slynch8** [comment]
thanks for the repro steps, I'll try and get to this soon
